### PR TITLE
Handle fragmented packets from tcp socket.

### PR DESCRIPTION
**Pull request by erdemaksu**
Use the length information to count the received bytes and relay complete packets to call back function.

### CODE REVIEW

File: lib/transport.js [hunk 1]
```diff
@@ -14,7 +14,20 @@ class Transport {
 
   connect(cb) {
     return this._openSocket().then(() => this._authenticate()).then(() => {
-      this.socket.on('data', cb)
+      var buffers = []
```
Comment (doganyazar) [inline lib/transport.js]: Use let instead of var

File: lib/transport.js [hunk 2]
```diff
@@ -14,7 +14,20 @@ class Transport {
...
+      var rem, len
+      this.socket.on('data', (chunk) => {
+	if (buffers[0] == null) {
+	    len = rem = chunk.readUInt32BE(0)
+	    chunk = chunk.slice(4)
```
Comment (doganyazar) [inline lib/transport.js]: Use splice instead of slice since creates a new array

File: lib/transport.js [hunk 3]
```diff
@@ -14,7 +14,20 @@ class Transport {
 
   connect(cb) {
     return this._openSocket().then(() => this._authenticate()).then(() => {
-      this.socket.on('data', cb)
+      var buffers = []
+      var rem, len
+      this.socket.on('data', (chunk) => {
```
Comment (doganyazar) [inline lib/transport.js]: This works great but I was assuming that transport.js would be protocol agnostic so instead I would do exact same thing in client.js. But then again if you are happy with this I do not mind :)

File: lib/transport.js (hunk 2 above)
Comment (erdemaksu) [reply lib/transport.js]: chunk is of type Buffer, and Buffer class does not have splice method.

File: lib/transport.js (hunk 3 above)
Comment (erdemaksu) [reply lib/transport.js]: I'll keep length header at transport level. Proprietary wire protocol and Apollo protocol will be handled on higher level.

File: lib/transport.js (hunk 1 above)
Comment (erdemaksu) [reply lib/transport.js]: OK.

**Comment by erdemaksu** [comment]
Hello Panpa!
Please check the second commit.

**Pull request merged**